### PR TITLE
draupnir-client --insecure flag

### PR DESCRIPTION
**Pull request by lawrencejones**
Allow specification of an `--insecure` flag to the draupnir client that
allows skipping certificate validation. Useful for accessing the API on
the draupnir machine via localhost, when you don't want to validate
certificates and know you're talking to the right endpoint.

### CODE REVIEW

File: cli/main.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ import (
 	"time"
 
 	"github.com/burntsushi/toml"
-	"github.com/gocardless/draupnir/client"
+	clientPkg "github.com/gocardless/draupnir/client"
```
Comment (hmac) [inline cli/main.go]: ha ha ha ha sad.

File: cli/main.go [hunk 2]
```diff
@@ -69,16 +69,31 @@ func main() {
...
+		cli.BoolFlag{
+			Name:  "insecure",
+			Usage: "don't validate certificates when connecting to draupnir",
+		},
+	}
+
+	// Construct a client before we run any of our commands
+	app.Before = func(c *cli.Context) error {
+		client = clientPkg.NewClient(
+			fmt.Sprintf("https://%s", CONFIG.Domain),
```
Comment (hmac) [inline cli/main.go]: Doesn't this need to be http for `--insecure`?

Comment (lawrencejones) [reply cli/main.go]: Nope, as the goal is to target the https endpoint but not check certs

File: client/client.go [hunk 3]
```diff
@@ -43,11 +58,11 @@ type DraupnirClient interface {
...
 
 func (c Client) AuthorizationHeader() string {
-	return fmt.Sprintf("Bearer %s", c.Token.RefreshToken)
+	return fmt.Sprintf("Bearer %s", c.token.RefreshToken)
 }
 
-func (c Client) get(url string) (*http.Response, error) {
-	req, err := http.NewRequest(http.MethodGet, url, strings.NewReader(""))
+func (c Client) get(path string) (*http.Response, error) {
+	req, err := http.NewRequest(http.MethodGet, c.url+path, strings.NewReader(""))
```
Comment (hmac) [inline client/client.go]: I would expect this to be something like
```go
c.client.Get(c.url+path)
```

File: cli/main.go (hunk 2 above)
Comment (hmac) [reply cli/main.go]: Cool, given that's the ultimate aim this makes sense.

File: client/client.go (hunk 3 above)
Comment (lawrencejones) [reply client/client.go]: Ah absolutely. That's my bad, of course this still worse.

Comment (hmac) [reply client/client.go]: Cool - the rest looks good; can you ping me when it's ready for a final review?

File: client/client.go [hunk 4]
```diff
@@ -318,6 +286,45 @@ func (c Client) CreateAccessToken(state string) (oauth2.Token, error) {
 	return token, err
 }
 
+func (c Client) do(req *http.Request) (*http.Response, error) {
```
Comment (lawrencejones) [inline client/client.go]: Moved this into a single `do` method to avoid the duplication.

Comment (hmac) [reply client/client.go]: Largely like this 👍

**Pull request merged**